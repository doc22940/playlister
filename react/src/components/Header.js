import React, { Component }  from 'react';
import { Link } from 'react-router-dom';

class Header extends Component {
  render() {
    return (
      <header>
        <nav className='navbar navbar-expand-sm bg-primary fixed-top'>
          <Link to='/' className='navbar-brand text-light'>Playlister</Link>
          <Link to='/old-404' className='nav-item text-light'>Funny</Link>
        </nav>
      </header>
    );
  }
}

export default Header;

