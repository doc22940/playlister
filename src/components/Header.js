import React, { Component }  from 'react';
import { NavLink, Link } from 'react-router-dom';

class Header extends Component {
  closeMenu() {
    try {
      document.getElementsByClassName('navbar-toggler')[0].click();
    } catch (err) {
      console.error('trying to close the navbar on mobile. Ignore.', err);
    }
  }

  render() {
    return (
      <header>
        <nav className='navbar navbar-expand-xs bg-dark navbar-dark fixed-top'>
          <Link to='/' className='navbar-brand text-light'>Playlister</Link>

          <div id='navbarResponsiveContent' className='collapse navbar-collapse'>
            <ul className='navbar-nav justify-content-center'>
              <NavLink
                to='/playlists'
                className='nav-item nav-link'
                onClick={this.closeMenu}>
                Playlists
              </NavLink>
              <NavLink
                to='/videos'
                className='nav-item nav-link'
                onClick={this.closeMenu}>
                Videos
              </NavLink>
            </ul>
          </div>

          <button className='navbar-toggler' data-toggle='collapse' data-target='#navbarResponsiveContent' aria-controls='navbarResponsiveContent' aria-expanded='false' aria-label='Toggle navigation'>
            <span className='navbar-toggler-icon'></span>
          </button>
        </nav>
      </header>
    );
  }
}

export default Header;

