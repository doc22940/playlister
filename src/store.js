import { createStore, applyMiddleware, compose } from 'redux';
import { routerMiddleware } from 'react-router-redux';
import createHistory from 'history/createBrowserHistory';
import createSagaMiddleware from 'redux-saga';

import rootReducer from './modules';
import sagas from './sagas';

export const history = createHistory();

const sagaMiddleware = createSagaMiddleware();
const initialState = {};
const enhancers = [];
const middleware = [routerMiddleware(history), sagaMiddleware];

if (process.env.NODE_ENV === 'development') {
  const { devToolsExtension } = window;

  if (typeof devToolsExtension === 'function') {
    enhancers.push(devToolsExtension());
  }
}

const composedEnhancers = compose(applyMiddleware(...middleware), ...enhancers);

const store = createStore(rootReducer, initialState, composedEnhancers);

sagaMiddleware.run(sagas);

export default store;

