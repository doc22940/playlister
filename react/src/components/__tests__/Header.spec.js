import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router';
import { Provider } from 'react-redux'

import Header from '../Header';
import store from '../../store';

test('Header snapshot', () => {
  const component = renderer.create(
    <Provider store={store}>
      <MemoryRouter>
        <Header />
      </MemoryRouter>
    </Provider>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

