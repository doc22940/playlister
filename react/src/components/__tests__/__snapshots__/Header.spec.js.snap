// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshot 1`] = `
<header>
  <nav
    className="navbar navbar-expand-sm bg-primary fixed-top"
  >
    <a
      className="navbar-brand text-light"
      href="/"
      onClick={[Function]}
    >
      Playlister
    </a>
  </nav>
</header>
`;
